### ISSUE TITLE
.WorkExternalIdentifierType required in v1.2 - but so far failed to find the right posting format...

### ISSUE THREAD

**Issue by romanchyla** [ORCID/ORCID-Source#1817]
I'm so far failing to find the right combination of values to successfully POST to /orcid-works. But maybe there is a bug with the type validation, any help is much appreciated

error:
```
Can not construct instance of org.orcid.jaxb.model.message.WorkExternalIdentifierType from String value 'bibcode': value not one of declared Enum instance names
 at [Source: org.apache.catalina.connector.CoyoteInputStream@1d48a4f5; line: 21, column: 18] (through reference chain: org.orcid.jaxb.model.message.OrcidMessage["orcid-profile"]->org.orcid.jaxb.model.message.OrcidProfile["orcid-activities"]->org.orcid.jaxb.model.message.OrcidActivities["orcid-works"]->org.orcid.jaxb.model.message.OrcidWorks["orcid-work"]->org.orcid.jaxb.model.message.OrcidWork["work-external-identifiers"]->org.orcid.jaxb.model.message.WorkExternalIdentifiers["work-external-identifier"]->org.orcid.jaxb.model.message.WorkExternalIdentifier["work-external-identifier-type"])
```

this is the datastruct i'm trying to post (full request below):

```
            "work-external-identifiers": {
              "work-external-identifier": [
                 {
                 "work-external-identifier-type": "bibcode",
                 "work-external-identifier-id": "1924MNRAS..84..308E."
                 }
              ],
            },
```

this is the list of accepted values: http://support.orcid.org/knowledgebase/articles/118807
I also checked wiht the xsd - the 'bibcode' is there in the list of allowed values

I also tried posting xml format, nope....

```
        <work-external-identifiers>
          <work-external-identifier>
            <work-external-identifier-type>﻿source-work-id</work-external-identifier-type>
            <work-external-identifier-id>1234</work-external-identifier-id>
          </work-external-identifier>
        </work-external-identifiers>
```

full request:

```
POST /v1.2/0000-0001-8178-9506/orcid-works HTTP/1.1
Host: api.sandbox.orcid.org
Content-length: 2593
Content-type: application/json
Authorization: Bearer 986b6d98-96ac-4350-89eb-8914e38f492b
Accept: application/json
{
  "message-version": "1.2", 
  "orcid-profile": {
    "group-type": null, 
    "client-type": null, 
    "orcid-identifier": {
      "path": "0000-0001-8178-9506", 
      "host": "sandbox.orcid.org", 
      "uri": "http://sandbox.orcid.org/0000-0001-8178-9506", 
      "value": null
    }, 
    "orcid-activities": {
      "affiliations": null, 
      "orcid-works": {
        "scope": null, 
        "orcid-work": [
          {
            "work-external-identifiers": {
              "work-external-identifier": [
                 {
                 "work-external-identifier-type": "bibcode",
                 "work-external-identifier-id": "1924MNRAS..84..308E."
                 }
              ],
            },

            "put-code": "466190", 
            "source": {
              "source-orcid": {
                "path": "0000-0001-8178-9506", 
                "host": "sandbox.orcid.org", 
                "uri": "http://sandbox.orcid.org/0000-0001-8178-9506", 
                "value": null
              }, 
              "source-name": {
                "value": "Roman Chyla"
              }, 
              "source-date": {
                "value": 1422645668284
              }
            }, 
            "work-title": {
              "subtitle": null, 
              "title": {
                "value": "ADS 2.0"
              }
            }, 
            "last-modified-date": {
              "value": 1422645668284
            }, 
            "created-date": {
              "value": 1422645668284
            }, 
            "visibility": "PUBLIC", 
            "work-type": "JOURNAL_ARTICLE", 
            "publication-date": {
              "month": {
                "value": "01"
              }, 
              "day": {
                "value": "01"
              }, 
              "media-type": null, 
              "year": {
                "value": "2015"
              }
            }, 
            "journal-title": {
              "value": "foo"
            }, 
            "url": null
          }
        ]
      }
    }, 
    "orcid": null, 
    "type": "USER", 
    "orcid-preferences": {
      "locale": "EN"
    }, 
    "orcid-history": {
      "last-modified-date": {
        "value": 1424116495004
      }, 
      "creation-method": "DIRECT", 
      "submission-date": {
        "value": 1422645321288
      }, 
      "verified-primary-email": {
        "value": false
      }, 
      "verified-email": {
        "value": false
      }, 
      "visibility": null, 
      "source": null, 
      "claimed": {
        "value": true
      }
    }
  }
}
HTTP/1.1 400 Bad Request
Via: HTTP/1.1 GWA
X-google-cache-control: remote-fetch
Server: nginx/1.1.19
Connection: keep-alive
Date: Mon, 16 Feb 2015 20:09:40 GMT
Access-control-allow-origin: *
Content-type: text/plain;charset=UTF-8
Can not construct instance of org.orcid.jaxb.model.message.WorkExternalIdentifierType from String value 'bibcode': value not one of declared Enum instance names
 at [Source: org.apache.catalina.connector.CoyoteInputStream@4f062489; line: 21, column: 18] (through reference chain: org.orcid.jaxb.model.message.OrcidMessage["orcid-profile"]->org.orcid.jaxb.model.message.OrcidProfile["orcid-activities"]->org.orcid.jaxb.model.message.OrcidActivities["orcid-works"]->org.orcid.jaxb.model.message.OrcidWorks["orcid-work"]->org.orcid.jaxb.model.message.OrcidWork["work-external-identifiers"]->org.orcid.jaxb.model.message.WorkExternalIdentifiers["work-external-identifier"]->org.orcid.jaxb.model.message.WorkExternalIdentifier["work-external-identifier-type"])
```

**Comment by rcpeters** [comment]
@lizkrznarich Please update the documentation. For json the enums must be uppercase. (fyi V2 of the api xml and json will be uppercase. @romanchyla use "BIBCODE" for json.

**Comment by romanchyla** [comment]
And instead of dashes there are underscores

**Comment by rcpeters** [comment]
Indeed. Beyond painful in my opinion. V2 is the first time I've personally
will have gotten any chance to make any major changes. In V2 xml and json
will have matching enums. You see parts of V2 in sandbox it next couple of
weeks.

Cheers,
Rob

On Fri, Feb 20, 2015 at 9:38 AM, Roman Chyla <notifications@github.com>
wrote:

> And instead of dashes there are underscores
> On 20 Feb 2015 12:35, "Robert Peters" <notifications@github.com> wrote:
>
> > Closed #1817 <https://github.com/ORCID/ORCID-Source/issues/1817>.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/ORCID/ORCID-Source/issues/1817#event-239278294>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ORCID/ORCID-Source/issues/1817#issuecomment-75284408>.
>



-- 
Robert Peters
info@rcpeters.com
805.440.9056
